src/port1.0/portbump.tcl

```diff
@@ -0,0 +1,234 @@
+                    set calculated_sum [portchecksum::calc_$type $fullpath]
+                    lappend calculated_checksums $type
+                    lappend calculated_checksums $calculated_sum
+                    lappend both_checksums $type $sum $calculated_sum
+
+                    if {$sum eq $calculated_sum} {
+                        ui_debug "[format [msgcat::mc "Correct (%s) bump for %s"] $type $distfile]"
+                    } else {
+                        ui_info "[format [msgcat::mc "Portfile bump: %s %s %s"] $distfile $type $sum]"
+                        ui_info "[format [msgcat::mc "Distfile bump: %s %s %s"] $distfile $type $calculated_sum]"
+
+                        # Raise the failure flag
+                        set mismatch yes
+                    }
+                }
+
+                # Save our calculated checksums in case we need them later
+                set calculated_checksums_array($distfile) $calculated_checksums
+
+                if {![regexp {\.html?$} ${distfile}] &&
+                    ![catch {strsed [exec [findBinary file $portutil::autoconf::file_path] $fullpath --brief --mime] {s/;.*$//}} mimetype]
+                    && "text/html" eq $mimetype} {
+                    # file --mime-type would be preferable to file --mime and strsed, but is only available as of Snow Leopard
+                    set wrong_mimetype yes
+                    set htmlfile_path ${fullpath}.html
+                    file rename -force $fullpath $htmlfile_path
+                }
+            }
+        }
+    } else {
+        # Something went wrong with the syntax.
+        return -code error "[msgcat::mc "Unable to verify file checksums"]"
+    }
+
+    if {![tbool mismatch]} {
+        ui_msg "No changes needed."
+        return 0
+    }
+
+    if {[tbool wrong_mimetype]} {
+        # We got an HTML file, though the distfile name does not suggest that one was
+        # expected. Probably a helpful DNS server sent us to its search results page
+        # instead of admitting that the server we asked for doesn't exist, or a mirror that
+        # no longer has the file served its error page with a 200 response.
+        ui_notice "***"
+        ui_notice "The non-matching file appears to be HTML. See this page for possible reasons"
+        ui_notice "for the bump mismatch:"
+        ui_notice "<https://trac.macports.org/wiki/MisbehavingServers>"
+        ui_notice "***"
+        ui_notice "The file has been moved to: $htmlfile_path"
+        
+        return -code error "[msgcat::mc "Unable to verify file checksums"]"
+    } else {
+        # Show the desired checksum line for easy cut-paste
+        # based on the previously calculated values, plus our default types
+        set sums {}
+
+        foreach distfile $all_dist_files {
+            if {[llength $all_dist_files] > 1} {
+                lappend sums $distfile
+            }
+
+            set missing_types $portchecksum::default_checksum_types
```

Comment:
I'm not too skilled in tcl, but this line failed for me. What worked was to keep just `$default_checksum_types` (remove `portchecksum`), but I had to add that to `global` variables on top. I don't know what the proper solution is though.

Reply:
I think it's the branch yeah, as it runs fine on my machine. I'll check it out later when the implementation is done. Can I resolve this for now?